utilities/backup/backup_engine.cc

```diff
@@ -595,8 +595,29 @@ class BackupEngineImpl {
                              Temperature file_temp, RateLimiter* rate_limiter,
                              std::string* db_id, std::string* db_session_id);
 
-  struct CopyOrCreateResult {
-    ~CopyOrCreateResult() {
+  struct WorkItemResult {
```

Comment:
It looks like the current plan is to have WorkItemResult and WorkItem contain all the fields for the various types of work item. This is probably ok for the foreseeable future but std::variant would likely be better if things become more fragmented.

Reply:
Good callout!